f8a_worker/storages/s3_manual_tagging.py

```diff
@@ -0,0 +1,28 @@
+from . import AmazonS3
+
+
+class S3ManualTagging(AmazonS3):
+    def store_user_data(self, input_json):
+        file_name = self.get_file_name(input_json)
+        data = input_json.get('data', {})
+        ecosystem = data.get('ecosystem', None)
+        if ecosystem != 'pypi':
+            file_name = ecosystem + "/" + file_name
+        self.store_dict(data, file_name)
+
+    def fetch_user_data(self, input_json):
+        user_json = {}
+        ecosystem = input_json.get('ecosystem', None)
+        file_name = self.get_file_name(input_json)
+        if ecosystem != 'pypi':
+            file_name = ecosystem + "/" + file_name
+        file_data = self.retrieve_dict(file_name)
+        user_json['user'] = input_json['user']
+        user_json['data'] = file_data
+        return user_json
+
+    def get_file_name(self, input_json):
+        user_name = input_json.get('user', None)
+        postfix = "_user.json"
+        file_name = user_name + postfix
```

Comment:
If user would be None, this will result in an error:
```python

>>> None + '_user.json'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

What about directly supplying user's name instead of input json? I think it will be less error prone.

Reply:
okay. let me do the changes